app/src/store/index.js

```diff
@@ -1,11 +1,63 @@
 import Vue from "vue";
 import Vuex from "vuex";
+// import axios from "axios";
+import Api from "./api";
 
 Vue.use(Vuex);
 
 export default new Vuex.Store({
-  state: {},
-  mutations: {},
-  actions: {},
+  state: {
+    apiBaseUrl: 'http://localhost:4000/',
+    currentUser: null,
+    header: {
+      "headers":
+      {
+        "content-type": "application/json",
+      }
+    }
+  },
+  // change the state tree
+  mutations: {
+    setCurrentUser(currentState, user) {
+      currentState.currentUser = user;
+    }
+  },
+  actions: {
+    login: (context, credentials) => {
+      return Api.post("login", credentials, context.state.header).then((jwt) => {
+        context.commit('setCurrentUser', jwt);
+        console.log(context.state.currentUser);
+      }).catch(error => { return Promise.reject(error); });
+    },
+
+    register: (context, credentials) => {
+      context.state.header = {
+        "headers":
+        {
+          "content-type": "application/json",
+          "Authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhY2Nlc3NfbGV2ZWwiOjEsImV4cCI6MTU3ODg3ODUzNywib3JpZ19pYXQiOjE1Nzg4NzQ5MzcsInV1aWQiOiI4YTM1YzMwYi00NmVkLTQyN2QtYWU4Yi1jNzQ3OTIzNjQ1MzYifQ.FHch528hinD7vd_wWh5pevj8sLjbJ8Ss437swYNZPU8"
```

Comment:
Je sais que n'a pas encore fini mais pour ça, faudrait ajouter le token en local storage pour ensuite l'utiliser dans toutes les autres requêtes en l'incluant dans le header de tous les calls api axios comme ceci : `axios.defaults.headers.common.Authorization = `Bearer ${authToken}`;`

Voici des références à ce que je veux dire : 
- https://github.com/axios/axios#config-defaults
- https://stackoverflow.com/questions/54652685/how-can-i-store-a-jwt-in-cookies-using-axios